### PR TITLE
[Filter/TF] refactor to init tf model

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. set in/out tensor info while loading and verifying tf model
2. copy raw data into tensor instead loop and remove macros to set tensor data

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc [hunk 1]
```diff
@@ -350,112 +371,35 @@ TFCore::run (const GstTensorMemory * input, GstTensorMemory * output)
...
-        copyInputWithType (float);
-        break;
-      case _NNS_FLOAT64:
-        copyInputWithType (double);
-        break;
-      default:
-        /** @todo Support other types */
-        break;
-    }
+    Tensor in (input_tensor_info[i].type, input_tensor_info[i].shape);
```
Comment (helloahn) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: I will revive `TensorShape` when the `TensorCApi::makeTensor()` is available in our system.

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc [hunk 2]
```diff
@@ -93,52 +89,57 @@ TFCore::getModelPath ()
...
  * @return 0 if OK. non-zero if error.
  *        -1 if the pb file is not loaded.
- *        -2 if the input properties is different with model.
+ *        -2 if the input properties are different with model.
  *        -3 if the Tensorflow session is not created.
  */
 int
 TFCore::loadModel ()
 {
+  g_message ("Start to load model [%s]", model_path);
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: Please don't write messages unconditionally for "normal operations".

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc [hunk 3]
```diff
@@ -93,52 +89,57 @@ TFCore::getModelPath ()
...
+  for (int i = 0; i < outputTensorMeta.num_tensors; ++i) {
+    output_tensor_names.push_back (outputTensorMeta.info[i].name);
+  }
+
 #if (DBG)
   gint64 stop_time = g_get_real_time ();
   g_message ("Model is loaded: %" G_GINT64_FORMAT, (stop_time - start_time));
 #endif
+
+  g_message ("Finished to load model.");
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: Make it GST_INFO

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc [hunk 4]
```diff
@@ -226,9 +190,25 @@ TFCore::getTensorTypeToTF (tensor_type tType)
  *        -5 if the rank of input tensors exceeds our capacity NNS_TENSOR_RANK_LIMIT.
  */
 int
-TFCore::inputTensorValidation (const std::vector <const NodeDef*> &placeholders)
+TFCore::validateInputTensor (GraphDef &graph_def)
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: can it be const? (graph_def)

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc [hunk 5]
```diff
@@ -271,31 +251,81 @@ TFCore::inputTensorValidation (const std::vector <const NodeDef*> &placeholders)
...
+
+/**
+ * @brief	check the inserted information about output tensor with model
+ * @return 0 if OK. non-zero if error.
+ *        -1 if the number of output tensors is not matched.
+ *        -2 if the dimension of output tensors is not matched.
+ *        -3 if the type of output tensors is not matched.
+ */
+int
+TFCore::validateOutputTensor (std::vector <Tensor> &outputs)
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: It'd be better if ```outputs``` becomes const.

**Comment by myungjoo** [comment]
Anyway, please do not come here during vacation. Enjoy!

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc (hunk 2 above)
Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: logs about start/end to load model are removed

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc (hunk 3 above)
Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: logs about start/end to load model are removed

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc (hunk 5 above)
Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: I changed the param to const.

File: ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc (hunk 4 above)
Comment (jaeyun-jung) [reply ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc]: I changed the param to const.

**Pull request merged**